tensorboard/plugins/debugger/debugger_server_lib.py

```diff
@@ -0,0 +1,321 @@
+               events_writer_manager,
+               numerics_alert_callback=None):
+    """Constructor of DebuggerDataStreamHandler.
+
+    Args:
+      events_writer_manager: Manages writing events to disk.
+      numerics_alert_callback: An optional callback run every time a health pill
+        event with bad values (Nan, -Inf, or +Inf) is received. The callback
+        takes the event as a parameter.
+    """
+    self._events_writer_manager = events_writer_manager
+    self._numerics_alert_callback = numerics_alert_callback
+
+    # We use session_run_index as the "step" value for debugger events because
+    # it is unique across all runs. It is not specific to a set of feeds and
+    # fetches.
+    self._session_run_index = -1
+
+  def on_core_metadata_event(self, event):
+    """Implementation of the core metadata-carrying Event proto callback.
+
+    Args:
+      event: An Event proto that contains core metadata about the debugged
+        Session::Run() in its log_message.message field, as a JSON string.
+        See the doc string of debug_data.DebugDumpDir.core_metadata for details.
+    """
+    self._session_run_index = self._parse_session_run_index(event)
+
+  def on_graph_def(self, graph_def, device_name, wall_time):
+    """Implementation of the GraphDef-carrying Event proto callback.
+
+    Args:
+      graph_def: A GraphDef proto. N.B.: The GraphDef is from
+        the core runtime of a debugged Session::Run() call, after graph
+        partition. Therefore it may differ from the GraphDef available to
+        the general TensorBoard. For example, the GraphDef in general
+        TensorBoard may get partitioned for multiple devices (CPUs and GPUs),
+        each of which will generate a GraphDef event proto sent to this
+        method.
+      device_name: Name of the device on which the graph was created.
+      wall_time: An epoch timestamp (in microseconds) for the graph.
+    """
+    # For now, we do nothing with the graph def. However, we must define this
+    # method to satisfy the handler's interface. Furthermore, we may use the
+    # graph in the future (for instance to provide a graph if there is no graph
+    # provided otherwise).
+    del device_name
+    del wall_time
+    del graph_def
+
+  def on_value_event(self, event):
+    """Records the summary values based on an updated message from the debugger.
+
+    Logs an error message if writing the event to disk fails.
+
+    Args:
+      event: The Event proto to be processed.
+    """
+    if not event.summary.value:
+      tf.logging.warn("The summary of the event lacks a value.")
+      return
+
+    # The node name property is actually a watch key, which is a concatenation
+    # of several pieces of data.
+    watch_key = event.summary.value[0].node_name
+    if not watch_key.endswith(constants.DEBUG_NUMERIC_SUMMARY_SUFFIX):
+      # Ignore events that lack a DebugNumericSummary.
+      # NOTE(chizeng): We may later handle other types of debug ops.
```

Comment:
Done. Replaced existing instances of those with github usernames, including @chihuahua's and mine.